src/unix/mediactrl.cpp

```diff
@@ -308,7 +313,12 @@ static gint gtk_window_realize_callback(GtkWidget* widget,
 
 #if GST_CHECK_VERSION(1,0,0)
     gst_video_overlay_set_window_handle(be->m_xoverlay,
+#ifdef GDK_WINDOWING_X11
                                 GDK_WINDOW_XID(window)
+#endif
+#ifdef GDK_WINDOWING_WAYLAND
+                                (guintptr)gdk_wayland_window_get_wl_surface(window)
+#endif
```

Comment:
This isn't going to work - `GDK_WINDOWING_X11` and `GDK_WINDOWING_WAYLAND` are compile-time checks and they can both be defined at the same time.  There needs to be a runtime check here too.

Also, this change isn't enough to make this work on Wayland (I've tried).

Reply:
I still have to investigate this: using Yocto Dunfell and Variscite DART-MX8M-MINI EVK, the patch give no compile errors.
Also the mediaplayer example code compiles just fine, but when starting it shows

root@imx8mm-var-dart:/usr/share/wxwidgets/samples/mediaplayer# ./mediaplayer                        
mediaplayer.cpp(1504): assert "bOK" failed in wxMediaPlayerNotebookPage(): Could not create media control!

Is there anywhere in the wxWidgets code a reference to see how the x11/wayland "runtime check" can be implemented?

Reply:
Yes, I can imagine it compiles fine when only `GDK_WINDOWING_WAYLAND` is defined, but both are defined on standard distributions of GTK3, so that case needs to be handled.

For the runtime checks, you can use ` GDK_IS_X11_WINDOW()` and `GDK_IS_WAYLAND_WINDOW()`.  See src/unix/glegl.cpp for an example of that.